torch/distributed/fsdp/sharded_grad_scaler.py

```diff
@@ -348,8 +350,12 @@ def update(self, new_scale: Optional[Union[float, torch.Tensor]] = None) -> None
             if isinstance(new_scale, float):
                 self._scale.fill_(new_scale)  # type: ignore[union-attr]
             else:
-                reason = "new_scale should be a float or a 1-element torch.cuda.FloatTensor with requires_grad=False."
-                assert isinstance(new_scale, torch.cuda.FloatTensor), reason  # type: ignore[attr-defined]
+                reason = "new_scale should be a float or a 1-element torch.cuda.FloatTensor or \
+                    torch.FloatTensor with requires_grad=False."
+                if self._device == "cuda":
+                    assert isinstance(new_scale, torch.cuda.FloatTensor), reason  # type: ignore[attr-defined]
+                else:
+                    assert isinstance(new_scale, torch.FloatTensor), reason  # type: ignore[attr-defined]
```

Comment:
Can you just modernize this test to not use the legacy constructors. `new_scale.device.type == self._device`, assuming `self._device` is always a string

Reply:
Do you mean to change it to this: `assert new_scale.device.type == self._device` ?

Reply:
Yup!

Reply:
Revised